sys/include/hashes/sha3.h

```diff
@@ -0,0 +1,190 @@
+
+/**
+ * @brief SHA-3-256 initialization.  Begins a SHA-3-256 operation.
+ *
+ * @param ctx  keccak_state_t handle to initialise
+ */
+void sha3_256_init(keccak_state_t *ctx);
+
+/**
+ * @brief Add bytes into the hash
+ *
+ * @param ctx      context handle to use
+ * @param[in] data Input data
+ * @param[in] len  Length of @p data
+ */
+void sha3_update(keccak_state_t *ctx, const void *data, size_t len);
+
+/**
+ * @brief SHA-3-256 finalization.  Pads the input data and exports the hash value
+ *
+ * @param ctx    context handle to use
+ * @param digest resulting digest, this is the hash of all the bytes
+ */
+void sha3_256_final(keccak_state_t *ctx, void *digest);
+
+/**
+ * @brief SHA-3-384 initialization.  Begins a SHA-3-256 operation.
+ *
+ * @param ctx  keccak_state_t handle to initialise
+ */
+void sha3_384_init(keccak_state_t *ctx);
+
+/**
+ * @brief SHA-3-384 finalization.  Pads the input data and exports the hash value
+ *
+ * @param ctx    context handle to use
+ * @param digest resulting digest, this is the hash of all the bytes
+ */
+void sha3_384_final(keccak_state_t *ctx, void *digest);
+
+/**
+ * @brief SHA-3-512 initialization.  Begins a SHA-3-256 operation.
+ *
+ * @param ctx  keccak_state_t handle to initialise
+ */
+void sha3_512_init(keccak_state_t *ctx);
+
+/**
+ * @brief SHA-3-512 finalization.  Pads the input data and exports the hash value
+ *
+ * @param ctx    context handle to use
+ * @param digest resulting digest, this is the hash of all the bytes
+ */
+
+void sha3_512_final(keccak_state_t *ctx, void *digest);
+
+/**
+ * @brief A wrapper function to simplify the generation of a hash, this is
+ * usefull for generating SHA-3-256 from one buffer
+ *
+ * @param[in] data     pointer to the buffer to generate hash from
+ * @param[in] len      length of the buffer
+ * @param[out] digest  pointer to an array for the result, length must
+ *                     be SHA3_256_DIGEST_LENGTH
+ */
+void sha3_256(void *digest, const void *data, size_t len);
+
+/**
+ * @brief A wrapper function to simplify the generation of a hash, this is
+ * usefull for generating SHA-3-384 from one buffer
+ *
+ * @param[in] data     pointer to the buffer to generate hash from
+ * @param[in] len      length of the buffer
+ * @param[out] digest  pointer to an array for the result, length must
+ *                     be SHA3_384_DIGEST_LENGTH
+ */
+void sha3_384(void *digest, const void *data, size_t len);
+
+/**
+ * @brief A wrapper function to simplify the generation of a hash, this is
+ * usefull for generating SHA-3-512 from one buffer
+ *
+ * @param[in] data     pointer to the buffer to generate hash from
+ * @param[in] len      length of the buffer
+ * @param[out] digest  pointer to an array for the result, length must
+ *                     be SHA3_512_DIGEST_LENGTH
+ */
+void sha3_512(void *digest, const void *data, size_t len);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif  /* HASHES_SHA3_H */
```

Comment:
I'm not sure what exactly what kind, but the `🚫⏎` at the end indicates some kind of whitespace issue (maybe a missing newline, but as I said, I'm not sure). Please fix.

Reply:
It means that there is no newline at the end of the file. Normally, text files on Unix always end with a newline, it makes them display correctly if printed to a text terminal, e.g. `cat filename`